src/ArrayCollection.php

```diff
@@ -65,6 +70,7 @@ class ArrayCollection implements Collection, Selectable, Stringable
     public function __construct(array $elements = [])
     {
         $this->elements = $elements;
+        $this->size     = empty($elements) ? 0 : count($elements);
```

Comment:
You don't need to avoid `count()` _on an empty array_. That's a cheap operation. `count`ing non-empty arrays preemptively is what troubles me.